SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosStatusUpdateHandler.java

```diff
@@ -344,6 +346,35 @@ private StatusUpdateResult unsafeProcessStatusUpdate(
           taskIdObj
         );
         return StatusUpdateResult.KILL_TASK;
+      } else if (
+        maybeTaskHistory.isPresent() &&
+        status.getReason() == Reason.REASON_AGENT_REREGISTERED
```

Comment:
I think this is already covered by the enclosing isRecoveryStatusUpdate if statement

Reply:
I don't see anything specific to re-registration in that method, but can definitely add this logic to run in either scenario